### PR TITLE
fix gcc 8.4.1 compilation issues

### PR DESCRIPTION
**Pull request by garlick**
gcc 8.4.1 on fluke found some legitimate problems.  Address them, and add some relaxations for the osu benchmarks which we should not modify unless necessary.

### CODE REVIEW

**Comment by grondo** [comment]
Any clue why these errors don't pop up in the centos8 builder in ci?
```
gcc (GCC) 8.4.1 20200928 (Red Hat 8.4.1-1)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

If we're missing some flags we should add those in this PR IMO.

**Comment by garlick** [comment]
Good point!  I'll dig into that tomorrow.  I was surprised to hit those for the first time on fluke, but then I got excited about getting everything compiled up for a test and didn't circle back.  Thanks for reminding me.

**Comment by grondo** [comment]
The problem seems to be that `WARNING_CFLAGS` is only defined in `configure.ac` if `ac_cv_c_compiler_vendor` is `gnu` or `clang`, but `ac_cv_c_compiler_vendor` is set from `AX_COMPILER_VENDOR`, which is called but `config/ax_compiler_vendor.m4` is missing.

This results in an empty `WARNING_CFLAGS`. So we have not had `-Wall` etc. enabled in CI.

Looks like `config/ax_compiler_version.m4` is also missing. If I add those then the build fails on `centos8` (and perhaps we'll find errors on other distros as well)

**Comment by garlick** [comment]
Thanks @grondo, I added those files in the last commit, and no other compiler errors popped up in CI.  Sorry for my sloppiness!

File: t/osu-micro-benchmarks/Makefile.am [hunk 1]
```diff
@@ -1,6 +1,11 @@
 AM_CFLAGS = \
         $(WARNING_CFLAGS) \
-        $(CODE_COVERAGE_CFLAGS)
+        $(CODE_COVERAGE_CFLAGS) \
+	-Wno-error=switch \
+	-Wno-error=format \
+	-Wno-error=format-extra-args \
+	-Wno-error=unused-variable \
+	-Wno-error=maybe-uninitialized
```
Comment (grondo) [inline t/osu-micro-benchmarks/Makefile.am:8]: Inconsistent spaces vs tabs

**Review by grondo — APPROVED** [review]
LGTM! One very minor whitespace nit inline.

**Comment by garlick** [comment]
I'll set MWP then.  Thanks for the assist!

**Pull request merged**